src/rules.py

```diff
@@ -36,3 +64,24 @@ def is_winner(self, board):
             return True
         else:
             return False
+
+    def is_row_array_same(self, row_array):
```

Comment:
this name is super technical, especially out of context.  Can it be renamed using domain language?

Reply:
What do you think of...`three_in_a_row`?  or would it be `is_three_in_a_row`?  I don't know if those are winners but something along those lines?

Reply:
The things that concerned me where, is that your implementation detail has leaked out into the function name.  Also that parameter `row_array`.

At a domain level, I think the question you are asking is: is this one the winning mark?

Assuming that's the question, I would suggest the name: `is_winning_mark?`